### PR TITLE
DEEP-9655 Add build id to package info

### PR DESCRIPTION
**Pull request by namandf**

### CODE REVIEW

File: pkg/golang/binary/parse.go [hunk 1]
```diff
@@ -57,11 +63,38 @@ func (p *Parser) Parse(r dio.ReadSeekerAt) ([]types.Library, []types.Dependency,
 			mod = dep.Replace
 		}
 
+		if !checkBuildID {
+			// get build id
+			buildID, err = getBuildID(r)
```
Comment (john-d8r) [inline pkg/golang/binary/parse.go]: Why not calculate this outside the loop ?

Comment (namandf) [reply pkg/golang/binary/parse.go]: Trivy skips binaries in the sbom which don't have any dep, so intend to skip computing build id in those cases

Comment (namandf) [reply pkg/golang/binary/parse.go]: Update

**Review by john-d8r — APPROVED** [review]
LGTM

**Pull request merged**